File: backend/app/presentation/schemas/moex.py
from typing import List, Optional

from pydantic import BaseModel


class MoexQuoteOut(BaseModel):
    ticker: str
    short_name: Optional[str] = None
    last: Optional[float] = None
    change: Optional[float] = None
    change_percent: Optional[float] = None
    update_time: Optional[str] = None
    currency: str = "RUB"


class MoexQuotesResponse(BaseModel):
    index: str
    source: str
    fetched_at: str
    fallback: bool = False
    items: List[MoexQuoteOut]

File: frontend/src/api/moex.ts
import { api } from "./client";

export type MoexQuote = {
  ticker: string;
  short_name: string | null;
  last: number | null;
  change: number | null;
  change_percent: number | null;
  update_time: string | null;
  currency: string;
};

export type MoexQuotesResponse = {
  index: string;
  source: string;
  fetched_at: string;
  fallback: boolean;
  items: MoexQuote[];
};

export async function getImoexQuotes(limit = 12): Promise<MoexQuotesResponse> {
  const res = await api.get<MoexQuotesResponse>("/public/moex/imoex/quotes", {
    params: { limit },
  });
  return res.data;
}
